### PR TITLE
Added `le` and `ge` bounds to constrained numerics.

### PR DESCRIPTION
**Pull request by jaheba**
See discussion #174.

### CODE REVIEW

**Comment by jaheba** [comment]
Probably we want to update the docs as well.

File: pydantic/types.py [hunk 1]
```diff
@@ -175,19 +175,34 @@ def validate(cls, value, values, **kwarg):
...
 
 
-class ConstrainedInt(int):
+class ConstrainedMeta(type):
+    def __new__(cls, name, bases, dct):
+        new_cls = type.__new__(cls, name, bases, dct)
+        try:
+            assert new_cls.gt is None or new_cls.ge is None, ('gt', 'ge')
+            assert new_cls.lt is None or new_cls.le is None, ('lt', 'le')
+        except AssertionError as exception:
```
Comment (Gr1N) [inline pydantic/types.py]:
```
except AssertionError as e:
   ...
   raise errors.ConfigError(...) from e
```
```

File: tests/test_types.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from pydantic import (DSN, UUID1, UUID3, UUID4, UUID5, BaseModel, EmailStr, NameEmail, NegativeFloat, NegativeInt,
                       PositiveFloat, PositiveInt, PyObject, StrictStr, ValidationError, condecimal, confloat, conint,
                       constr, create_model)
+from pydantic.errors import ConfigError
```
Comment (Gr1N) [inline tests/test_types.py]:
```
from pydantic import ConfigError
```

File: pydantic/types.py [hunk 3]
```diff
@@ -175,19 +175,34 @@ def validate(cls, value, values, **kwarg):
         return make_dsn(**kwargs)
 
 
-class ConstrainedInt(int):
+class ConstrainedMeta(type):
```
Comment (Gr1N) [inline pydantic/types.py]: Maybe better to change name of class? Something like `ConstrainedNumberMeta` or just `NumberMeta`?

**Comment by jaheba** [comment]
Addressed all comments, I think.

File: docs/examples/exotic.py [hunk 4]
```diff
@@ -16,10 +16,12 @@ class Model(BaseModel):
     strip_str: constr(strip_whitespace=True)
 
     big_int: conint(gt=1000, lt=1024) = None
+    # inclusive: conint(ge=1000, le=1024)
```
Comment (samuelcolvin) [inline docs/examples/exotic.py]: remove

**Comment by jaheba** [comment]
Anything else?

**Comment by samuelcolvin** [comment]
great, thank you very much.

**Pull request merged**